### PR TITLE
Run the AddToReset transform at the beginning

### PR DESCRIPTION
**Pull request by ryanhall07**
https://github.com/m3db/m3/pull/3668 introduced a bug where the
AddToReset transform was only run before updating the aggregations map.
This resulted in constant misses when looking up from the aggregations
map when determining if the metadata should be updated.

Fortunately an idempotent check in the addNewAggregation function
prevented duplicate aggregations from getting created. This did have the
unfortunate side effect of creating a new aggregationValues slice
unnecessarily per metric ingested.

**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/aggregator/aggregator/entry.go [hunk 1]
```diff
@@ -646,9 +646,6 @@ func (e *Entry) updateStagedMetadatasWithLock(
 
 	// Update the metadatas.
 	for _, p := range sm.Pipelines {
-		if e.opts.AddToReset() {
```
Comment (ryanhall07) [inline src/aggregator/aggregator/entry.go:649]: essentially we were transforming the map key only when adding, not when looking from the map (which happens here https://github.com/m3db/m3/blob/264b9eb34ae55a19ad44cf846a4e30cb93578c33/src/aggregator/aggregator/entry.go#L534)

Comment (ryanhall07) [reply src/aggregator/aggregator/entry.go:649]: and we were efficiently allocating a new slice per metric ingested here

https://github.com/m3db/m3/blob/264b9eb34ae55a19ad44cf846a4e30cb93578c33/src/aggregator/aggregator/entry.go#L644

Comment (ryanhall07) [reply src/aggregator/aggregator/entry.go:649]: and this idempotent check prevented any logic bugs (i.e creating duplicate aggs)

https://github.com/m3db/m3/blob/264b9eb34ae55a19ad44cf846a4e30cb93578c33/src/aggregator/aggregator/entry.go#L583

File: src/aggregator/aggregator/aggregator.go [hunk 2]
```diff
@@ -281,6 +283,16 @@ func (agg *aggregator) AddTimedWithStagedMetadatas(
 	return nil
 }
 
+func (agg *aggregator) updateStagedMetadatas(sms metadata.StagedMetadatas) {
```
Comment (ryanhall07) [inline src/aggregator/aggregator/aggregator.go:286]: just update the metadata initially when the request is received.

**Review by wesleyk — APPROVED** [review]

**Review by robskillington — APPROVED** [review]
LGTM

**Pull request merged**